### PR TITLE
scrape data

### PR DESCRIPTION
**Pull request by wipedlife**

### CODE REVIEW

File: src/tracker/scrape/scrape.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package scrape;
+
+import (
+	b "github.com/jackpal/bencode-go"
```
Comment (wipedlife) [inline src/tracker/scrape/scrape.go]: I think to use https://github.com/zeebo/bencode instead it now for that

**Comment by majestrate** [comment]
can you run `go fmt` on the code?

**Comment by wipedlife-nohome** [comment]
I did it.

File: src/tracker/scrape/scrape.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import (
+	"fmt"
+	b "github.com/felix/go-bencode"
+
+	"crypto/sha1"
+	"io"
+	"os"
+)
+
+func FilesConstructMap(raw interface{}) map[string]map[string]int64 {
```
Comment (majestrate) [inline src/tracker/scrape/scrape.go]: you can break this up by making a struct that will de-serialize data into when decoding

File: src/tracker/scrape/getFile.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+	src := []byte(hash)
+	bin := make([]byte, hex.DecodedLen(len(src)))
+
+	n, err = hex.Decode(bin, src)
+	if err != nil {
+		return err, nil
+	}
+	info_hash := fmt.Sprintf("%s", net_url.QueryEscape(string(bin[:n])))
+	if err := DownloadFile(filepath+"_"+hash, url+"?info_hash="+info_hash); err != nil {
```
Comment (majestrate) [inline src/tracker/scrape/getFile.go:43]: `url` is not escaped, should it be?

**Pull request merged**